src/o365/spo/commands/file/file-get.ts

```diff
@@ -0,0 +1,277 @@
+        }
+
+        let requestUrl: string;
+        let options: string;
+
+        if (args.options.id) {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileById('${encodeURIComponent(args.options.id)}')`;
+        }
+        else {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(args.options.url)}')`;
+        }
+
+        if (args.options.asListItem) {
+          options = '?$expand=ListItemAllFields';
+        }
+        else if (!args.options.asListItem && !args.options.asFile && !args.options.asString) {
+          options = '';
+        }
+        else {
+          options = '/$value';
+        }
+
+        const requestOptions: any = {
+          url: requestUrl + options,
+          method: 'GET',
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'accept': 'application/json;odata=nometadata'
+          }),
+          encoding: null, // Set encoding to null, otherwise binary data will be encoded to utf8 and binary data is corrupt 
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((file: string): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(file);
+          cmd.log('');
+        }
+        
+        if (args.options.asString) {
+          cmd.log(file);
+        }
+        else if (args.options.asListItem) {
+          let fileProperties: FileProperties = JSON.parse(JSON.stringify(file));
+          cmd.log(fileProperties.ListItemAllFields)
+        }
+        else if(args.options.asFile) {
+          this.writeFile(file, args.options.path, args.options.fileName)
+          cmd.log(`File ${args.options.fileName} saved to path ${args.options.path}`);
+        }
+        else {
+          let fileProperties: FileProperties = JSON.parse(JSON.stringify(file));
+          cmd.log(fileProperties);
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  private writeFile(fileContent: string, filePath: string, fileName: string): void {
+    let fullPath: string = path.join(filePath, fileName);
+
+    fs.writeFileSync(fullPath, fileContent);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-w, --webUrl <webUrl>',
+        description: 'The URL of the site where the folder from which to retrieve files is located'
+      },
+      {
+        option: '-u, --url [url]',
+        description: 'server- or site-relative URL of the file. Specify either url or id but not both'
+      },
+      {
+        option: '-i, --id [id]',
+        description: 'file ID. Specify either url or id but not both'
+      },
+      {
+        option: '--asString',
```

Comment:
When I try to get a file as string, I'm getting the following output:

```text
0  : 239
1  : 187
10 : 101
100: 101
101: 49
102: 49
103: 49
104: 101
105: 57
106: 52
107: 50
...
```

This doesn't seem to be correct. Are you by any chance passing the whole buffer into the output without converting it into a string first?